### ISSUE TITLE
Start-Process -Wait doesn't work when running as a different user (-Credential)

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#17033]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

On Windows, when you combine `Start-Process -Wait` with `-Credential` (running as a different user), the process is launched, BUT
* an `Access denied` error is reported
* no waiting occurs.

By contrast, using `-PassThru` and `.WaitForExit()` on the passed through `System.Diagnostics.Process` instance works fine.

```powershell
# Run on Windows
# Be sure to specify valid credentials when prompted.
Start-Process -Credential (Get-Credential) cmd  -Wait
```

### Expected behavior

No error should occur, and the call should block until the launched `cmd` instance is exited.

### Actual behavior

The `cmd` instance is launched, but, instead of waiting, an `Access denied` error is emitted.

Contrast this with this seemingly equivalent command, which works as expected:

```powershell
(Start-Process -Credential (Get-Credential) cmd -PassThru).WaitForExit()
```

### Environment data

```powershell
PowerShell Core 7.3.0-preview.2 on Windows
```

**Comment by jborean93** [comment]
The problem here is that PowerShell is calling `CreateProcessWithLogon` directly and then getting the [dotnet Process object](https://github.com/PowerShell/PowerShell/blob/8ad642013a5495ce102d9857afb4dcb9a44dc00f/src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs#L2571-L2572) manually. Dotnet will attempt to open a new handle on that process but it will fail causing the problem you see here. Unfortunately I'm not aware of any public way to create the `Process` object from a handle returned by the various `CreateProcess` APIs. I'm not actually sure why .NET calls the API directly instead of going through the .NET process itself but there might be that I'm not aware off.

I see some potential fixes for this

* Have PowerShell use .NET to spawn the process with the credentials so the Process object it gets back has full control over the object and can be waited on
* Have PowerShell return the process handle from it's internal calls and use that in the wait job builder
* Petition .NET to create a public API that can construct a Process object from an IntPtr/SafeHandle that PowerShell provides
* Don't use PowerShell for this task

The first is probably the easiest to implement and would simplify the code. I'm just not sure why it's there in the first place so there potentially might be a scenario with `Start-Process` that is not achievable through .NET. The 2nd is also doable but doesn't help the `-PassThru` scenario as the process object you get back will be under the same limitations you see today. It will be enough to wait on but some actions won't be available. I have no idea how feasible the 3rd option is, I've certainly wanted something like this before but it's a Windows specific feature and might require a lot of work to expose properly there. The 4th doesn't really help PowerShell but certainly an option to get your code working today.

**Comment by iSazonov** [comment]
There were many limitations in .Net Framework and .Net Core Process class implementation. As a result, we have a low-level implementation that comes from Windows PowerShell. Now we could review new implementation in .Net 6+. I think we can move to .Net implementation with some minor breaking changes. Perhaps we could resolve many issues and requested enhancements with Start-Process in the time. Unfortunately, I did not find support earlier when I tried to do this job.
